public-ui/src/app/data-browser/views/survey-chart/survey-chart-react.component.tsx

```diff
@@ -0,0 +1,182 @@
+import { AgeChartReactComponent } from 'app/data-browser/charts/chart-age/chart-age-react.component';
+import { BioSexChartReactComponent } from 'app/data-browser/charts/chart-biosex/chart-biosex-react.component';
+import { VersionChartReactComponent } from 'app/data-browser/charts/chart-version/chart-version-react.component';
+import { TooltipReactComponent } from 'app/data-browser/components/tooltip/tooltip-react.component';
+import { ErrorMessageReactComponent } from 'app/data-browser/views/error-message/error-message-react.component';
+import { GraphType } from 'app/utils/enum-defs';
+import { triggerEvent } from 'app/utils/google_analytics';
+import * as React from 'react';
+
+interface State {
+    graphToShow: string;
+    selectedAnalysisId: number;
+    selectedChartAnalysis: any;
+    displayGraphErrorMessage: boolean;
+    isLoaded: boolean;
+}
+
+interface Props {
+    graphButtons: any;
+    question: any;
+    answer: any;
+    selectedResult: any;
+    surveyName: any;
+    searchTerm: any;
+    surveyCountAnalysis: any;
+    isCopeSurvey: any;
+    versionAnalysis: any;
+}
+
+export class SurveyChartReactComponent extends React.Component<Props, State> {
+  constructor(props) {
+    super(props);
+    this.state = {
+        graphToShow: this.props.isCopeSurvey ? GraphType.SurveyVersion : GraphType.BiologicalSex,
+        displayGraphErrorMessage: false,
+        isLoaded: false,
+        selectedChartAnalysis: null,
+    };
+  }
+
+  componentDidMount() {
+    this.selectGraphType(this.state.graphToShow, this.props.question, this.props.answer);
+  }
+
+  selectGraphType(g: any, q: any, answer: any) {
+    const {surveyName, searchTerm} = this.props;
+    if (answer.stratum4.toLowerCase().indexOf('more than one race') > -1) {
+      triggerEvent('conceptClick', 'More than one race /ethncitiy graph view',
+        'Expand to see graphs', surveyName + ' - Q'
+        + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
+        ' - ' + g, searchTerm, null);
+    }
+    triggerEvent('conceptClick', 'View Graphs',
+      'Expand to see graphs', surveyName + ' - Q'
+      + q.actualQuestionNumber + ' - ' + q.conceptName + ' - ' + answer.stratum4 +
+      ' - ' + g, searchTerm, null);
+    q.graphToShow = g;
+    switch (g) {
+        case GraphType.AgeWhenSurveyWasTaken:
+            q.selectedAnalysis = q.ageAnalysis;
+            break;
+        case GraphType.SurveyVersion:
+            q.selectedAnalysis = q.versionAnalysis;
+            break;
+        default:
+            q.selectedAnalysis = q.genderAnalysis;
+            break;
+    }
+    this.setState({
+        graphToShow: g,
+        selectedChartAnalysis: q.selectedAnalysis,
+        isLoaded: true,
+        displayGraphErrorMessage: q.selectedAnalysis === undefined ||
+        (q.selectedAnalysis && q.selectedAnalysis.results.filter(a => a.stratum3 ===
+        answer.stratum3).length === 0) });
```

Comment:
I think modifying `q` here is causing the other answer graphs to change. Since `q` is the `question` prop and is passed to this function directly in `render`, any changes to it will get passed up to the parent through props, then down to the other charts.

I made these changes and seemed to at least be showing different data on each chart:
```suggestion
    let selectedAnalysis;
    switch (g) {
        case GraphType.AgeWhenSurveyWasTaken:
            selectedAnalysis = q.ageAnalysis;
            break;
        case GraphType.SurveyVersion:
            selectedAnalysis = q.versionAnalysis;
            break;
        default:
            selectedAnalysis = q.genderAnalysis;
            break;
    }
    this.setState({
        graphToShow: g,
        selectedChartAnalysis: selectedAnalysis,
        isLoaded: true,
        displayGraphErrorMessage: selectedAnalysis === undefined ||
        (selectedAnalysis && selectedAnalysis.results.filter(a => a.stratum3 ===
        answer.stratum3).length === 0) });
```

Reply:
Thanks, it also does not seem to flipping the charts with other chart button click.